superset-frontend/plugins/plugin-chart-echarts/src/Histogram/transformProps.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { BarSeriesOption, EChartsOption } from 'echarts';
+import { CallbackDataParams } from 'echarts/types/src/util/types';
+import { isEmpty } from 'lodash';
+import {
+  CategoricalColorNamespace,
+  NumberFormats,
+  getColumnLabel,
+  getNumberFormatter,
+  tooltipHtml,
+} from '@superset-ui/core';
+import { HistogramChartProps, HistogramTransformedProps } from './types';
+import { LegendOrientation, LegendType, Refs } from '../types';
+import { defaultGrid, defaultYAxis } from '../defaults';
+import { getLegendProps } from '../utils/series';
+import { getDefaultTooltip } from '../utils/tooltip';
+import { getPercentFormatter } from '../utils/formatters';
+
+const TOTAL_SERIES = 'total';
+
+export default function transformProps(
+  chartProps: HistogramChartProps,
+): HistogramTransformedProps {
+  const refs: Refs = {};
+  let focusedSeries: number | undefined;
+  const {
+    formData,
+    height,
+    hooks,
+    legendState = {},
+    queriesData,
+    theme,
+    width,
+  } = chartProps;
+  const { onLegendStateChanged } = hooks;
+  const {
+    colorScheme,
+    column,
+    groupby = [],
+    normalize,
+    showLegend,
+    showValue,
+    sliceId,
+    xAxisTitle,
+    yAxisTitle,
+  } = formData;
+  const { data } = queriesData[0];
+  const colorFn = CategoricalColorNamespace.getScale(colorScheme);
+  const formatter = getNumberFormatter(
```

Comment:
Should we also support currency formatter here?

Reply:
Let me give you some context to answer this question. Now that we're migrating legacy plugins, it becomes clear that we have a lot of duplicated code and many inconsistencies between ECharts plugins regarding the controls that are available. I'm planning to tackle this horizontally, making sure that features, such as currency formatting, are available consistently among the plugins. For that reason, for this PR:

> I added the minimum required controls to allow the migration but many more controls can be added in the future to improve the chart's customizability.

Reply:
Sorry, missed that part of the pr description. Makes sense!

Reply:
No problem! You correctly spotted the missing control.